ludwig/utils/data_utils.py

```diff
@@ -190,12 +198,20 @@ def read_xsv(data_fp, df_lib=PANDAS_DF, separator=",", header=0, nrows=None, ski
     if nrows is not None:
         kwargs["nrows"] = nrows
 
+    # If fetching a preview, create an iterator that will immediately be used to read the first chunk:
+    if "get_preview" in kwargs:
```

Comment:
Same as in `inner_method`, you could get rid of this if block with `kwargs["chunksize"] = kwargs.pop("get_preview", None)`.